docs/reference/tags/experience_dates_search_tag/index.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: default
+title: Experience dates search
+parent: Tags
+grand_parent: Reference
+---
+
+# Experience dates search
+
+For additional examples see [here]({% link docs/experience_dates_search/index.md %}).
+
+The experience dates search tag executes a search on the company's product dates, it then paginates those results.
+
+##### input
+{% raw %}
+```liquid
+{% experience_dates_search product_id: 'abcd1234-1234-abcd-1234-abcd1234abcd', duration: { greater_than: 3, less_than: 8 }, page_size: 12, sort: 'departure_date_asc' %}
+    {% for item in result.items %}
+        <p>{{ item.name }}</p>
+    {% endfor %}
+    <div>
+        {{ paginate.collection_size }} Results
+        {{ paginate | default_pagination }}
+    </div>
+{% endexperience_dates_search %}
+```
+{% endraw %}
+
+##### output
+{% raw %}
+```html
+<p>Beyond Sahara</p>
+<p>Beyond Archipelago</p>
+<p>Beyond Sahara Friends & Family</p>
+<p>Beyond Archipelago Friends & Family</p>
+<p>Beyond Asia</p>
+<div>
+    112 Results
+    <span class="prev">
+    <a href="http://beyondadventures.com/pagination-test?page=2">Previous</a>
+    </span>
+    <span class="page">
+    <a href="http://beyondadventures.com/pagination-test?page=1">1</a>
+    </span>
+    <span class="page">
+    <a href="http://beyondadventures.com/pagination-test?page=2">2</a>
+    </span>
+    <span class="page current">3</span>
+    <span class="page">
+    <a href="http://beyondadventures.com/pagination-test?page=4">4</a>
+    </span>
+    <span class="page">
+    <a href="http://beyondadventures.com/pagination-test?page=5">5</a>
+    </span>
+    <span class="deco">&hellip;</span>
+    <span class="page">
+    <a href="http://beyondadventures.com/pagination-test?page=9">9</a>
+    </span>
+    <span class="next">
+    <a href="http://beyondadventures.com/pagination-test?page=4">Next</a>
+    </span>
+</div>
+```
+{% endraw %}
+
+The tag returns an `items` array and a `Paginate` pagination object to the block passed in, the items array represents 1 page of the results from the search.
+
+
+## Pagination
+
+The results of this search are paginated for performance reasons, with this you can define the number of results to display per page as the `page_size` attribute, which defaults to 12.
```

Comment:
I wonder if we need to provide a way to paginate by month, instead of by page. Basically I think it would need to accept a `month` and `year`.

I had an example of that in the [RFC: Pagination on Sites with Turbo Frames](https://www.notion.so/easol/RFC-Pagination-on-Sites-with-Turbo-Frames-6d024a2493ed4d2caf62227bcf3d8d30?pvs=4#9e9263ae999b4a9f88f8e54426d4ed95).

Full code example can be seen on this branch - [ian/slots-on-sites-without-frame-tag](https://github.com/easolhq/easol/compare/main...ian/slots-on-sites-without-frame-tag?expand=1)

Reply:
I actually think it would be better to use different options for the two ways to paginate:

1. By page: use `page: page_num` and `page_size: 12`
2. By month: use `month: "2"` and `year: "2023"` (the year can default to the current year if missing or next year if the `month` has already occurred)

I'm not sure how the `paginate` object will be used for the latter.

Reply:
Perhaps we expose a different pagination, e.g. `month_pagination` or `calendar_pagination` 🤔 

I don't know what properties are currently being used on the `pagination`. Is it only being used with `default_pagination ` to generate the page links? If so, maybe we don't need it for months, as these links wouldn't be used on a calendar?